processor/spanmetricsprocessor/processor_test.go

```diff
@@ -493,3 +496,76 @@ func TestBuildKey(t *testing.T) {
 
 	assert.NotEqual(t, k0, k1)
 }
+
+func TestValidateDimensions(t *testing.T) {
+	for _, tc := range []struct {
+		name        string
+		dimensions  []Dimension
+		hasError bool
+		expectedErr error
+	}{
+		{
+			name: "no additional dimensions",
+			dimensions: []Dimension{},
+			hasError: false,
+		},
+		{
+			name: "no duplicate dimensions",
+			dimensions: []Dimension{
+				{Name: "http.service_name"},
+				{Name: "http.status_code"},
+			},
+			hasError: false,
+		},
+		{
+			name: "duplicate dimension with reserved labels",
+			dimensions: []Dimension{
+				{Name: "service.name"},
+			},
+			hasError: true,
+			expectedErr: errors.New("duplicate dimension name service.name"),
+		},
+		{
+			name: "duplicate dimension with reserved labels after sanitization",
+			dimensions: []Dimension{
+				{Name: "service_name"},
+			},
+			hasError: true,
+			expectedErr: errors.New("duplicate dimension name service_name after sanitization"),
+		},
+		{
+			name: "duplicate additional dimensions",
+			dimensions: []Dimension{
+				{Name: "service_name"},
+				{Name: "service_name"},
+			},
+			hasError: true,
+			expectedErr: errors.New("duplicate dimension name service_name sanitization"),
+		},
+		{
+			name: "duplicate additional dimensions after sanitization",
+			dimensions: []Dimension{
+				{Name: "http.status_code"},
+				{Name: "http_status_code"},
+			},
+			hasError: true,
+			expectedErr: errors.New("duplicate dimension name http_status_code sanitization"),
+		},
+		{
+			name: "we skip the case if the dimension name is the same after sanitization",
+			dimensions: []Dimension{
+				{Name: "http_status_code"},
+			},
+			hasError: false,
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
+			err := validateDimensions(tc.dimensions)
+			if tc.hasError {
+				assert.Error(t, err, tc.expectedErr)
```

Comment:
Please consider using `assert.EqualError` and replace `expectedErr error` with `expectedErr string`. I suspect this assertion is passing, but on a different error to the `expectedErr`.